### PR TITLE
[ios] Fix image padding for buttons in right-to-left layout

### LINKED ISSUE #7976: Alignment text
⚠ Have you searched for similar, already existing issues?
Did not find anything similar to.
**Describe the issue**
The text in photo need to be a little bit aligned.

**Steps to reproduce**
None.

**Screenshots**

image

**System information:**
 - Operating system and its version: [iOS 17.5]
 - Organic Maps version: [⁦2024.03.31-7⁩]
 - Device Model: [iPhone 13 Pro Max]

**Additional context**
That's it for now.
Thanks!

### PR DESCRIPTION
**Pull request by andre1110**
Fixes #7976 for buttons when an RTL script is used
Before | After
------ | -----
sss | ppp

### CODE REVIEW

File: iphone/Maps/Classes/CustomViews/NavigationDashboard/Views/RoutePreview/RoutePreviewStatus/BaseRoutePreviewStatus.swift [hunk 1]
```diff
@@ -84,7 +84,19 @@ final class BaseRoutePreviewStatus: SolidTouchView {
     }
   }
 
+  @objc static func setImagePadding(_ padding: CGFloat, for button: UIButton) {
```
Comment (kirylkaveryn) [inline iphone/Maps/Classes/CustomViews/NavigationDashboard/Views/RoutePreview/RoutePreviewStatus/BaseRoutePreviewStatus.swift:87]: Such kind of static methods from the some class should not be used for modifying the instances of another classes.

An extension for the UIButton should be created:
`UIButton+SetImagePadding.swift`

File: iphone/Maps/Classes/CustomViews/NavigationDashboard/Views/RoutePreview/RoutePreviewStatus/BaseRoutePreviewStatus.swift [hunk 2]
```diff
@@ -84,7 +84,19 @@ final class BaseRoutePreviewStatus: SolidTouchView {
...
+    let isRightToLeft = UIView.userInterfaceLayoutDirection(for: button.semanticContentAttribute) == .rightToLeft
```
Comment (kirylkaveryn) [inline iphone/Maps/Classes/CustomViews/NavigationDashboard/Views/RoutePreview/RoutePreviewStatus/BaseRoutePreviewStatus.swift:88]: `UIApplication.sharedApplication.userInterfaceLayoutDirection == .rightToLeft` will be enough

File: iphone/Maps/Classes/CustomViews/NavigationDashboard/Views/RoutePreview/RoutePreviewStatus/BaseRoutePreviewStatus.swift [hunk 3]
```diff
@@ -84,7 +84,19 @@ final class BaseRoutePreviewStatus: SolidTouchView {
...
+      button.contentEdgeInsets = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: padding)
+      button.imageEdgeInsets = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: -2 * padding)
+    } else {
+      button.contentEdgeInsets = UIEdgeInsets(top: 0, left: padding, bottom: 0, right: 0)
+      button.imageEdgeInsets = UIEdgeInsets(top: 0, left: -2 * padding, bottom: 0, right: 0)
+    }
+  }
+
   private func configManageRouteButton(_ button: UIButton) {
+    BaseRoutePreviewStatus.setImagePadding(8, for: button)
```
Comment (kirylkaveryn) [inline iphone/Maps/Classes/CustomViews/NavigationDashboard/Views/RoutePreview/RoutePreviewStatus/BaseRoutePreviewStatus.swift:99]:
```suggestion
    button.setImagePadding(8)
```

File: iphone/Maps/Classes/CustomViews/NavigationDashboard/Views/RoutePreview/MWMRoutePreview.mm [hunk 4]
```diff
@@ -179,11 +180,13 @@ - (void)setDrivingOptionsState:(MWMDrivingOptionsState)state {
   }];
 
   if (state == MWMDrivingOptionsStateDefine) {
+    [MWMBaseRoutePreviewStatus setImagePadding:0.0 for:self.drivingOptionsButton];
```
Comment (kirylkaveryn) [inline iphone/Maps/Classes/CustomViews/NavigationDashboard/Views/RoutePreview/MWMRoutePreview.mm:183]:
```suggestion
    [self.drivingOptionsButton setImagePadding(0.0)];
```

**Review by kirylkaveryn — CHANGES_REQUESTED** [review]

File: iphone/Maps/Classes/CustomViews/NavigationDashboard/Views/RoutePreview/RoutePreviewStatus/BaseRoutePreviewStatus.swift (hunk 2 above)
Comment (andre1110) [reply iphone/Maps/Classes/CustomViews/NavigationDashboard/Views/RoutePreview/RoutePreviewStatus/BaseRoutePreviewStatus.swift:88]: I don't agree with this one. If we gonna put code this into a global extension, it should be as generic as possible. sharedApplication doesn't care if semanticContentAttribute is explicitly set on some UIView down the ui tree

**Comment by andre1110** [comment]
@kirylkaveryn I pushed a commmit with the suggested changes, except for the one with `sharedApplication.userInterfaceLayoutDirection`. If you think it's still necessary for some reason, let me know

File: iphone/Maps/Classes/CustomViews/NavigationDashboard/Views/RoutePreview/RoutePreviewStatus/BaseRoutePreviewStatus.swift (hunk 2 above)
Comment (kirylkaveryn) [reply iphone/Maps/Classes/CustomViews/NavigationDashboard/Views/RoutePreview/RoutePreviewStatus/BaseRoutePreviewStatus.swift]: Ok! I accept this argument

**Review by kirylkaveryn — APPROVED** [review]

**Comment by kirylkaveryn** [comment]
Thanks!

**Pull request merged**